src/generators/Minecraft.res

```diff
@@ -117,6 +118,25 @@ module Cuboid = {
       rotate: 0.0,
     }
 
+    // When a face is flipped both vertically and horizontally, this is the same as rotating 180 degrees.
+    let flip = ({rectangle, flip, rotate}: t, f: Generator_Texture.flip) => {
+      rectangle: rectangle,
+      flip: flip == #None ? f : #None,
+      rotate: flip == #None ? rotate : flip != f ? rotate +. 180.0 : rotate,
+    }
+
+    let rotate = ({rectangle, flip, rotate}: t, r: float) => {
+      let (x, y, w, h) = rectangle
+
+      {
+        rectangle: mod_float(r +. 360.0, 180.0) == 90.0
```

Comment:
I'm not quite following what's happening here, but it looks like it's trying to adjust for an incorrect position, width and height?

Can you add a few comments explaining the problem and solution?

This appears to be based on 90 degree angles? If that's the case, would it be worth restricting it to just those rotations? e.g.

```
type rotation = [#deg90, #deg180, #deg270]

let rotate = ({rectangle, flip, rotate}: t, r: rotation) => {
```

Thanks

Reply:
I investigated a little further and I think I see the problem now.

I see that our cubiod points works fine for flip operations, but don't work for rotate operations, which only became apparent when introducing the center option.

This is a good workaround, and I am happy for this to go in exactly as-is for now.

Let's have a think about how to tackle the overall problem, looks like the current strategy of creating cubiods could be improved.